xbmc/pvr/PVRGUIActions.cpp

```diff
@@ -926,6 +926,30 @@ namespace PVR
     return true;
   }
 
+  bool CPVRGUIActions::PlayEpgTag(const CFileItemPtr &item, bool bPlayMinimized, bool bCheckResume) const
+  {
+      const CEpgInfoTagPtr epgTag(CPVRItem(item).GetEpgInfoTag());
+      if (!epgTag)
+        return false;
+
+      CStringPropertyMapPtr properties(new CStringPropertyMap());
+      std::string stream = epgTag->GetStreamUrl(properties);
+      if (stream.empty())
+      {
+        return false;
+      }
+
+      item->SetPath(stream);
+      for (auto const &entry : *properties.get()) {
+        item->SetProperty(entry.first, entry.second);
+      }
+
+      if (!bCheckResume || CheckResumeRecording(item))
+        CApplicationMessenger::GetInstance().PostMsg(TMSG_MEDIA_PLAY, 0, 0, static_cast<void*>(new CFileItem(*item)));
```

Comment:
I guess this has to be tweeked a little, but sorry no time atm, will take a closer look later. Basic idea is to have exactly one method in CPVRGUIActions calling this. I just merged a larger refactoring PR to achieve this...

Quick hint: you are missing for instance parental check here, but adding it here is wrong, we have one core "check and play" method here.